webrender/src/frame_builder.rs

```diff
@@ -302,59 +276,283 @@ impl FrameBuilder {
         prim_index
     }
 
-    pub fn notify_waiting_for_root_stacking_context(&mut self) {
-        self.has_root_stacking_context = false;
-    }
-
     pub fn push_stacking_context(
         &mut self,
-        reference_frame_offset: &LayerVector2D,
         pipeline_id: PipelineId,
         composite_ops: CompositeOps,
         transform_style: TransformStyle,
         is_backface_visible: bool,
         is_pipeline_root: bool,
+        clip_and_scroll: ClipAndScrollInfo,
```

Comment:
So one potential issue with this is that different items in a stacking context can have different clip nodes and even different scroll nodes.

Reply:
Yep, that's expected and handled - items in a picture are primitive runs with their own clip and scroll. This parameter just identifies the current clip / scroll nodes when the stacking context is pushed - it's used for plane splitting and a few other things.

Reply:
To expand on that, it effectively identifies a root reference frame for that stacking context.